server/notifications_slack.go

```diff
@@ -352,24 +353,43 @@ func sendTenantSlackNotifications(s *Server, tenantName string, msgs []Notificat
 		m := msgs[i]
 		err := sendMessageToSlackChannel(slackIntegration, m.NotificationMsg.Title, m.NotificationMsg.Message)
 		if err != nil {
-			s.Errorf("failed to send slack notification: " + err.Error())
-			var rateLimit *slack.RateLimitedError
-			if errors.As(err, &rateLimit) {
-				time.Sleep(rateLimit.RetryAfter)
-				// when we hit rate limit there's no point in trying to send further messages for this tenant
-				// instead retry them on the next run
-				return
+			if err.Error() == "channel_not_found" {
+				s.Warnf("[tenant: %v]failed to send slack notification: %v", tenantName, err.Error())
+			} else {
+				s.Errorf("[tenant: %v]failed to send slack notification: %v", tenantName, err.Error())
```

Comment:
Let's log this as an error only in case the error is not rate limit otherwise this should be warning